src/mgr/DaemonMetricCollector.cc

```diff
@@ -0,0 +1,138 @@
+#include "DaemonMetricCollector.h"
+#include "common/admin_socket_client.h"
+#include "include/common_fwd.h"
+#include "common/perf_counters.h"
+
+#include <string>
+#include <iostream>
+#include <filesystem>
+#include <regex>
+#include <boost/json/src.hpp>
+
+using json_object = boost::json::object;
+using json_value = boost::json::value;
+using json_array = boost::json::array;
```

Comment:
ok, this is the first time i've seen boost.json used in ceph. i tend to think we should stick with rapidjson, but boost.json is probably a lot easier to use for a prototype 👍